reframe/core/meta.py

```diff
@@ -81,12 +80,11 @@ def __getitem__(self, key):
                         # available in the current class' namespace.
                         for b in self['_rfm_bases']:
                             if key in b._rfm_var_space:
-                                v = b._rfm_var_space[key]
-                                v.__set_name__(self, key)
-
                                 # Store a deep-copy of the variable's
                                 # value and return.
-                                self._namespace[key] = v.default_value
+                                self._namespace[key] = (
+                                    b._rfm_var_space[key].default_value
+                                )
```

Comment:
Just a style thing (unless it's again too long):

```suggestion
                                v = b._rfm_var_space[key].default_value
                                self._namespace[key] = v
```